framework/include/interfaces/Coupleable.h

```diff
@@ -155,6 +164,15 @@ class Coupleable
    */
   const ADVariableValue & adCoupledValue(const std::string & var_name, unsigned int comp = 0) const;
 
+  /**
+   * Returns value of a coupled lower-dimensional variable for use in Automatic Differentiation
+   * @param var_name Name of coupled variable
+   * @param comp Component number for vector of coupled variables
+   * @return Reference to a ADVariableValue for the coupled variable
+   */
+  const ADVariableValue & adCoupledValueLower(const std::string & var_name,
+                                              unsigned int comp = 0) const;
```

Comment:
@lindsayad Could we handle `lower` via `adCoupledValue` by passing a parameter? We are doing that for API: 

```
  virtual void onBoundary(const Elem * elem,
                           unsigned int side,
                           BoundaryID bnd_id,
                           const Elem * lower_d_elem = nullptr) override;
```

I think consistency is important.